Onboarding.md

```diff
@@ -1,115 +1,174 @@
+
+In addition to the above cli tools, you will need to install the following to build and run the `ml-api-adapter`:
+
+
+###  1.1. <a name='macOS'></a>macOS
+```bash
+#none - you have everything you need!
 ```
 
-##### Ubuntu
-To install Linuxbrew, follow these [instructions](http://linuxbrew.sh/#install-linuxbrew)
+###  1.2. <a name='Linux'></a>Linux
+
+[todo]
 
-#### Installing Docker
-To install Docker, follow these instructions: [Docker for Mac](https://docs.docker.com/docker-for-mac/), [Docker for Ubuntu](https://docs.docker.com/install/linux/docker-ce/ubuntu/#install-using-the-repository)
+###  1.3. <a name='Windows'></a>Windows
 
+[todo]
 
-#### Installing Postman
-Please, follow these instructions: [Get Postman](https://www.getpostman.com/postman)
 
-Alternatively on **Ubuntu** you may run:
+
+##  2. <a name='InstallingandBuilding'></a>Installing and Building
+
+Firstly, clone your fork of the `ml-api-adapter` onto your local machine:
+```bash
+git clone https://github.com/<your_username>/ml-api-adapter.git
+```
+
+Then `cd` into the directory and install the node modules:
+```bash
+cd ml-api-adapter
+npm install
 ```
-wget https://dl.pstmn.io/download/latest/linux64 -O postman.tar.gz
-sudo tar -xzf postman.tar.gz -C /opt
-rm postman.tar.gz
-sudo ln -s /opt/Postman/Postman /usr/bin/postman
+
+> If you run into problems running `npm install`, make sure to check out the [Common Errors/FAQs](#CommonErrorsFAQs) below.
+
+
+##  3. <a name='RunningLocally'></a>Running Locally
+
+While you can get the `ml-api-adapter` up and running by itself, it isn't all that useful for testing. `ml-api-adapter` requires at a minimum:
+* `central-ledger`
+* `kafka`
+
+Follow the [onboarding guide](https://github.com/mojaloop/central-ledger/blob/master/Onboarding.md) for `central-ledger` to get the required local environment set up before running the following:
+
+Once you have the `central-ledger` up and running, start the `ml-api-adapter` with:
+```bash
+npm run start
 ```
 
-##### Setup Postman
-* open *Postman*
-* click **Import** and then **Import File**
-* navigate to the central_ledger directory and select [postman.json](./postman.json)
+> Alternatively, try running the `ml-api-adapter` with `docker-compose` to make the setup a little easier: [Running Inside Docker](#RunningInsideDocker).
 
-#### nvm
-If you don't have cURL already installed, on **Ubuntu** run `sudo apt install curl`
 
-Download the nvm install script via cURL:
+##  4. <a name='RunningInsideDocker'></a>Running Inside Docker
+
+We use `docker-compose` to manage and run the `ml-api-adapter` along with its dependencies with one command.
+
+>*Note:* Before starting all of the containers however, start the `mysql` container alone, to give it some more time to set up the necessary permissions (this only needs to be done once). This is a short-term workaround because the `ml-api-adapter` doesn't retry it's connection to MySQL.
+
+```bash
+docker-compose up mysql #first time only
```

Comment:
Perhaps add additional information here to clarify why this is required `first time only`

Reply:
Yep. It's included in the above note, but maybe that's not clear enough.